lib/remote/apilistener-configsync.cpp

```diff
@@ -258,6 +297,28 @@ Value ApiListener::ConfigDeleteObjectAPIHandler(const MessageOrigin::Ptr& origin
 		return Empty;
 	}
 
+	auto isObjectLocked = [&ptype, &objName] {
+		auto type = ApiListener::m_ObjectNames.find(ptype.get());
+		if (type == ApiListener::m_ObjectNames.end()) {
+			return false;
+		}
+
+		return type->second.find(objName) != type->second.end();
+	};
+
+	std::unique_lock<std::mutex> lock(m_ObjectNamesMutex);
+	if (isObjectLocked()) {
+		m_ObjectNamesCV.wait(lock, [&isObjectLocked]{ return !isObjectLocked(); });
+	}
+	// Add object name to the locked list to block all other threads that try to process a message affecting the same object.
+	m_ObjectNames[ptype.get()].emplace(objName);
+
+	Defer unblockAndNotifyOne([&ptype, &objName]{
```

Comment:
Isn't _lock_ just for the mapping? Do you have to hold it during the whole delete operation?